Turbulence model updates

tkarna opened the pull request:
Turbulence model updates:

- Vertical integration uses 2nd order DIRK22 scheme
- Fix stability functions
- Fix turbulence initialization
- Update turbulence parameters
- Update examples

stephankramer commented on thetis/stability_functions.py:
```diff
@@ -99,41 +112,51 @@ def __init__(self, lim_alpha_shear=True, lim_alpha_buoy=True,
+        self.nb1 = 0.0
+        self.nb2 = 0.0
+
+    def compute_alpha_shear_steady(self, ri_st, analytical=True):
```
Does this need a docstring rather than a description in comments?

tkarna replied:
yes, fixed.

stephankramer commented on thetis/stability_functions.py:
```diff
@@ -185,25 +187,64 @@ def cost(a_shear):
-    def compute_cmu0(self):
+    def compute_cmu0(self, analytical=True):
         """
         Computes the paramenter c_mu_0 from stability function parameters
```
Is this docstring still correct/complete? What does `analytical` do?

stephankramer commented on thetis/stability_functions.py:
```diff
@@ -185,25 +187,64 @@ def cost(a_shear):
         """
         Computes von Karman constant from the Psi Schmidt number.
 
         n, c1, c2 are GLS model parameters.
```
and cmu0 presumably?

tkarna replied:
docstrings fixed here too

stephankramer commented on thetis/turbulence.py:
```diff
@@ -682,7 +700,9 @@ def postprocess(self):
             set_func_min_val(self.viscosity, o.visc_min)
             set_func_min_val(self.diffusivity, o.diff_min)
 
+            # print_output(' -----')
```
I presume this is some frequently used debugging output? Can this go into a `if self._debug_output:` block?

tkarna replied:
Yes. Now moved to `print_debug` function.

tkarna commented:
Thanks for your comments, I updated the docstrings and cleaned up the code. While at it also remove the now obsolete `use_smooth_eddy_viscosity` option.
